File: joevideolib/src/main/java/ffmpeg/Ffmpeg.java
package ffmpeg;

import android.util.Log;

import com.arthenica.ffmpegkit.FFmpegKit;
import com.arthenica.ffmpegkit.ReturnCode;
import com.arthenica.ffmpegkit.SessionState;
import com.joe.joevideolib.BuildConfig;

import VideoHandle.CmdList;
import utils.CmdUtils;

public class Ffmpeg {

    public static boolean DEBUG = BuildConfig.DEBUG;
    private static final String TAG = "Ffmpeg";

    /**
     * 开始处理
     *
     * @param cmd              命令
     * @param onEditorListener 回调接口
     */
    public static void execCmd(CmdList cmd, final OnFfmpegProcessCallback onEditorListener) {
        String[] cmds = cmd.toArray(new String[0]);
        String command = CmdUtils.join(cmds);
        if (DEBUG) {
            Log.d(TAG, "FFmpeg command: " + command);
        }
        FFmpegKit.executeAsync(command, session -> {
            if (DEBUG) {
                Log.d(TAG, "completeCallback " + session);
                StringBuilder logs = new StringBuilder();
                for (com.arthenica.ffmpegkit.Log x : session.getAllLogs()) {
                    String s = x.getMessage();
                    logs.append(s);
                }
                Log.d(TAG, logs.toString());
            }
            SessionState state = session.getState();
            ReturnCode returnCode = session.getReturnCode();

            if (ReturnCode.isSuccess(returnCode)) {
                onEditorListener.onSuccess();
            } else {
                onEditorListener.onFailure(null);
            }
        }, log -> {
            if (DEBUG) {
                Log.d(TAG, log.getMessage());
            }
            Long duration = FfmpegLogParser.parseTime(log.getMessage());
            if (duration != null) {
                onEditorListener.onProgress(duration);
            }

        }, statistics -> {

        });
    }
}

File: joevideolib/src/main/java/ffmpeg/Ffmpeg.kt
package ffmpeg

import VideoHandle.CmdList
import android.util.Log
import com.arthenica.ffmpegkit.FFmpegKit
import com.arthenica.ffmpegkit.FFmpegSession
import com.arthenica.ffmpegkit.ReturnCode
import com.arthenica.ffmpegkit.Statistics
import com.joe.joevideolib.BuildConfig
import ffmpeg.FfmpegLogParser.parseTime
import utils.CmdUtils

object Ffmpeg {
    var DEBUG = BuildConfig.DEBUG
    private const val TAG = "Ffmpeg"

    /**
     * 开始处理
     *
     * @param cmd              命令
     * @param onEditorListener 回调接口
     */
    fun execCmd(cmd: CmdList, onEditorListener: OnFfmpegProcessCallback) {
        val cmds = cmd.toTypedArray()
        val command = CmdUtils.join(cmds)
        if (DEBUG) {
            Log.d(TAG, "FFmpeg command: $command")
        }
        FFmpegKit.executeAsync(command, { session: FFmpegSession ->
            if (DEBUG) {
                Log.d(TAG, "completeCallback $session")
                val logs = StringBuilder()
                for (x in session.allLogs) {
                    val s = x.message
                    logs.append(s)
                }
                Log.d(TAG, logs.toString())
            }
            val state = session.state
            val returnCode = session.returnCode
            if (ReturnCode.isSuccess(returnCode)) {
                onEditorListener.onSuccess()
            } else {
                onEditorListener.onFailure(null)
            }
        }, { log: com.arthenica.ffmpegkit.Log ->
            if (DEBUG) {
                Log.d(TAG, log.message)
            }
            val duration = parseTime(log.message)
            if (duration != null) {
                onEditorListener.onProgress(duration)
            }
        }) { statistics: Statistics? -> }
    }
}